core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,147 @@
+use std::{future::Future, time::Duration};
+
+use anyhow::Error;
+use chrono::Utc;
+use tokio::sync::watch;
+use zksync_config::DADispatcherConfig;
+use zksync_da_layers::DataAvailabilityInterface;
+use zksync_dal::{Connection, ConnectionPool, Core, CoreDal};
+use zksync_types::L1BatchNumber;
+
+use crate::metrics::METRICS;
+
+#[derive(Debug)]
+pub struct DataAvailabilityDispatcher {
+    client: Box<dyn DataAvailabilityInterface>,
+    pool: ConnectionPool<Core>,
+    config: DADispatcherConfig,
+}
+
+impl DataAvailabilityDispatcher {
+    pub fn new(
+        pool: ConnectionPool<Core>,
+        config: DADispatcherConfig,
+        client: Box<dyn DataAvailabilityInterface>,
+    ) -> Self {
+        Self {
+            pool,
+            config,
+            client,
+        }
+    }
+
+    pub async fn run(self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        let pool = self.pool.clone();
+        loop {
+            let mut conn = pool.connection_tagged("da_dispatcher").await.unwrap();
+
+            if *stop_receiver.borrow() {
+                tracing::info!("Stop signal received, da_dispatcher is shutting down");
+                break;
+            }
+
+            if let Err(err) = self.dispatch(&mut conn).await {
```

Comment:
Why are errors swallowed here?

Reply:
my idea was that if e.g. DA layer's node stops responding, and returns errors, it should break the whole `da_dispatcher` task, but this way we will still log all the errors

do you think that the errors should be propagated?